src/role-helpers.js

```diff
@@ -69,16 +69,54 @@ function getImplicitAriaRoles(currentNode) {
     }
   }
 
+  // <form /> with an accessible name?
+  if (currentNode.matches('form')) {
```

Comment:
This is just restoring the old implementation that did not care about accessible name.

Technically `<form>` does not have the `role="form"` because the accessible name is missing (don't ask me why. I haven't tested it yet with AT; just following the spec). We could apply some heuristics to guess if it actually has one but if we want to be strict we would need to run accessible name computation (expensive).

Assuming I finish `byRole('form', { name: 'some-name' })` we could leave this quirk and then ignore any element returned `getImplicitAriaRoles` if the accessible name is empty (and potentially warn if a form is queried without providing a name).